python/lsst/daf/fmt/mysql/mysqlStorage.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python
+
+#
+# See COPYRIGHT file at the top of the source tree
+#
+
+from lsst.daf.persistence import (Storage, StorageInterface,
+                                  NoRepositoryAtRoot, RepositoryCfg)
+from contextlib import closing
+import copy
+import MySQLdb
+import os
+import urlparse
+
+
+class MySqlStorage(StorageInterface):
+    """Defines the interface for a connection to a Storage location.
+
+    Requires that a .my.cnf file be present in the home directory
+    (``~/.my.cnf``) with the user and password to log into the database.
+
+    Each database represents a single repository.
+
+    Parameters
+    ----------
+    uri : string
+        A URI to connect to a mysql database storage location. The form of the
+        URI is:
+        `mysql://[host][:port][/[database]]`
+        For example:
+        mysql://localhost:3306/myRepository
+
+        URI or path that is used as the storage location.
+    create : bool
+        If True the  a new repository should be created at the root location.
+        If False then a new repository will not be created.
+
+    Raises
+    ------
+    NoRepositoryAtRoot
+        If create is False and a repository does not exist at the root
+        specified by uri then NoRepositoryAtRoot is raised.
+    """
+
+    @staticmethod
+    def registerFormatter(formattable, formatter):
+        if formattable in MySqlStorage.formatters:
+            raise RuntimeError("Registration of second formatter {} for " +
+                               "formattable class {}".format(formatter,
+                                                             formattable))
+        MySqlStorage.formatters[formattable] = formatter
+
+    formatters = {}
+
+    def __init__(self, uri, create):
+        self.mysql = MySQLdb.connect(read_default_file="~/.my.cnf")
```

Comment:
Hmm, here you connect to mysql using only `my.cnf` and ignoring `uri` which means that `my.cnf` must contain everyhting and `uri` is just a duplicate of what's in ` my.cnf`?